src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,249 @@
+import { compare } from "../../common/string/compare";
+import { litLocalizeLiteMixin } from "../../mixins/lit-localize-lite-mixin";
+
+interface CommandItem {
+  text: string;
+  domain: string;
+  service: string;
+  serviceData?: {};
+  onClick(ev: Event): void;
+}
+
+@customElement("ha-quick-open-dialog")
+export class QuickOpenDialog extends litLocalizeLiteMixin(LitElement) {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _commandItems: CommandItem[] = [];
+
+  @internalProperty() private _itemFilter = "";
+
+  @internalProperty() private _opened = false;
+
+  @internalProperty() private _commandMode = false;
+
+  public translationFragment = "quick-open";
+
+  public async showDialog(params: QuickOpenDialogParams) {
+    this._commandMode = params.commandMode || false;
+    this._opened = true;
+  }
+
+  public closeDialog() {
+    this._opened = false;
+    this._itemFilter = "";
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected updated(changedParams) {
+    super.updated(changedParams);
+    if (changedParams.has("resources")) {
+      this._commandItems = this._generateCommandItems();
+    }
+  }
+
+  protected render() {
+    if (!this._opened) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog .heading=${true} open @closed=${this.closeDialog} hideActions>
+        <paper-input
+          dialogInitialFocus
+          no-label-float
+          slot="heading"
+          class="heading"
+          @value-changed=${this._entityFilterChanged}
+          .label=${this.localize("ui.panel.quick-open.filter_placeholder")}
+          type="search"
+          value=${this._commandMode ? `>${this._itemFilter}` : this._itemFilter}
+        ></paper-input>
+        <mwc-list>
+          ${this._commandMode
+            ? this.renderCommandsList()
+            : this.renderEntityList()}
+        </mwc-list>
+      </ha-dialog>
+    `;
+  }
+
+  protected renderCommandsList() {
+    const items = this._filterCommandItems(
+      this._commandItems,
+      this._itemFilter
+    );
+
+    return html`
+      ${items.map(
+        ({ text, domain, service, serviceData, onClick }) => html`
+          <mwc-list-item
+            @click=${onClick}
+            .domain=${domain}
+            .service=${service}
+            .serviceData=${serviceData}
+            graphic="icon"
+          >
+            <ha-icon .icon=${domainIcon(domain)} slot="graphic"></ha-icon>
+            ${text}
+          </mwc-list-item>
+        `
+      )}
+    `;
+  }
+
+  protected renderEntityList() {
+    const entities = this._filterEntityItems(
+      Object.keys(this.hass.states),
+      this._itemFilter
+    );
+
+    return html`
+      ${entities.map((entity_id) => {
+        const domain = computeDomain(entity_id);
+        return html`
+          <mwc-list-item
+            @click=${this._entityMoreInfo}
+            .domain=${domain}
```

Comment:
Why do we need the domain?

Reply:
Ah. We don't need it there. Only later for the icon. Removed